types/api/stacks/transaction.ts

```diff
@@ -27,6 +27,7 @@ export type UnsignedStacksTransation = {
   pendingTxs: StxMempoolTransactionData[];
   memo?: string;
   isNFT?: boolean;
+  sponsored?: boolean;
```

Comment:
I think we also need to pass sponsored as a parameter to the function call `generateUnsignedContractCall` down below in the function `generateUnsignedTransaction`